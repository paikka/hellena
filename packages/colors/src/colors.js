/**
 * Copyright Paikka. 2019, 2019
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

export const black = '#000000';

export const white = '#ffffff';

export const primary100 = '#CCDDED';
export const primary200 = '#99BCDB';
export const primary300 = '#3993DA';
export const primary400 = '#166BC3';
export const primary500 = '#166BC3';
export const primary600 = '#166BC3';
export const primary = {
  100: primary100,
  200: primary200,
  300: primary300,
  400: primary400,
  500: primary500,
  600: primary600
};

export const success100 = '#9CE6D6';
export const success200 = '#6BD9C1';
export const success300 = '#3ACDAD';
export const success400 = '#07A07F';
export const success500 = '#04604C';
export const success600 = '#034033';
export const success = {
  100: success100,
  200: success200,
  300: success300,
  400: success400,
  500: success500,
  600: success600
};

export const danger100 = '#FCCFD5';
export const danger200 = '#FBA0AB';
export const danger300 = '#F97787';
export const danger400 = '#E01C3A';
export const danger500 = '#B30A27';
export const danger600 = '#88021C';
export const danger = {
  100: danger100,
  200: danger200,
  300: danger300,
  400: danger400,
  500: danger500,
  600: danger600
};
