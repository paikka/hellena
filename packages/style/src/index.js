/**
 * Copyright Paikka. 2019, 2019
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

export * from './utilities/variables';
export * from './utilities/functions';
