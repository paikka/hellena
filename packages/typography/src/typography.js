/**
 * Copyright Paikka. 2019, 2019
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

export const baseFontSize = '16px';

export const fontSize1 = '.5rem';
export const fontSize2 = '.875rem';
export const fontSize3 = '1rem';
export const fontSize4 = '1.125rem';
export const fontSize5 = '1.25rem';
export const fontSize6 = '1.5rem';
export const fontSize7 = '1.75rem';
export const fontSize8 = '2rem';
export const fontSize9 = '2.25rem';
export const fontSize10 = '2.625rem';
export const fontSize11 = '3rem';
export const fontSize12 = '3.375rem';
export const fontSize13 = '3.75rem';
export const fontSize14 = '4.25rem';
export const fontSize15 = '4.75rem';
export const fontSize = {
  1: fontSize1,
  2: fontSize2,
  3: fontSize3,
  4: fontSize4,
  5: fontSize5,
  6: fontSize6,
  7: fontSize7,
  8: fontSize8,
  9: fontSize9,
  10: fontSize10,
  11: fontSize11,
  12: fontSize12,
  13: fontSize13,
  14: fontSize14,
  15: fontSize15
};

export const fontWeight1 = 100;
export const fontWeight2 = 200;
export const fontWeight3 = 300;
export const fontWeight4 = 400;
export const fontWeight5 = 500;
export const fontWeight6 = 600;
export const fontWeight7 = 700;
export const fontWeight8 = 800;
export const fontWeight9 = 900;
export const fontWeight = {
  1: fontSize1,
  2: fontSize2,
  3: fontSize3,
  4: fontSize4,
  5: fontSize5,
  6: fontSize6,
  7: fontSize7,
  8: fontSize8,
  9: fontSize9
};

export const text1 = fontSize1;
export const text2 = fontSize2;
export const text3 = fontSize3;
export const text4 = fontSize4;
export const text = {
  1: fontSize1,
  2: fontSize2,
  3: fontSize3,
  4: fontSize4
};

export const heading1 = fontSize5;
export const heading2 = fontSize6;
export const heading3 = fontSize7;
export const heading4 = fontSize8;
export const heading5 = fontSize9;
export const heading6 = fontSize10;
export const heading = {
  1: fontSize5,
  2: fontSize6,
  3: fontSize7,
  4: fontSize8,
  5: fontSize9,
  6: fontSize10
};
