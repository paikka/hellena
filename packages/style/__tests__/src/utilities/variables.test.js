/**
 * Copyright Paikka. 2019, 2019
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { variables, fontSizes, fontWeights, headings } from '../../../src';

test('variables', () => {
  expect(variables).toMatchSnapshot();
});

test('Font size object assignment', () => {
  const expected = {
    1: '.5rem',
    2: '.875rem',
    3: '1rem',
    4: '1.125rem',
    5: '1.25rem',
    6: '1.5rem',
    7: '1.75rem',
    8: '2rem',
    9: '2.25rem',
    10: '2.625rem',
    11: '3rem',
    12: '3.375rem',
    13: '3.75rem',
    14: '4.25rem',
    15: '4.75rem'
  };
  expect(fontSizes).toEqual(expected);
});

test('Font weight object assignment', () => {
  const expected = {
    1: 300,
    2: 400,
    3: 600,
    4: 700,
    5: 800
  };
  expect(fontWeights).toEqual(expected);
});

test('Text and heading sizing object assignment', () => {
  const keys = [
    'text-1',
    'text-2',
    'text-3',
    'text-4',
    'text-5',
    'heading-1',
    'heading-2',
    'heading-3',
    'heading-4',
    'heading-5',
    'heading-6',
    'heading-7',
    'heading-8',
    'heading-9',
    'heading-10'
  ];
  // Faz o mapeamento e verifica a presen√ßa de todas as chaves
  keys.map(function(value, index, array) {
    expect(headings).toHaveProperty(value);
  });
});
