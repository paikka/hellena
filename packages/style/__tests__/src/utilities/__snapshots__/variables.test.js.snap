// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`variables 1`] = `undefined`;
